### DISCUSSION TITLE
Module not found: Can't resolve '@/components

### DISCUSSION THREAD

**Discussion by zhangwei900808** [vercel/next.js#27666] [Help]
I add jsconfig.json
```
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@/components/*": [
        "src/components/*"
      ]
    }
  }
}
```

but when I run is error 
```
error - ./src/components/context/mainContext/index.js:3:0
Module not found: Can't resolve '@/components/layout/pageHeader/logoInHeader'
  1 | import React, { useState, useEffect } from "react";
  2 | import { useRouter } from "next/router";
> 3 | import LogoInHeader from "@/components/layout/pageHeader/logoInHeader";
  4 | import Menus from "../../navMenus";
  5 | import Record from "../../layout/record";
  6 | import styles from "./index.module.scss";

```

![image](https://user-images.githubusercontent.com/4385154/127789508-ee20d7a0-9219-4c05-83ab-3d80852ea3f8.png)

**Comment by zhangwei900808** [comment]
who can help me

**Comment by polRk** [comment]
```json
{
  "compilerOptions": {
    "baseUrl": "src",
    "paths": {
      "@/components/*": ["components/*"]
    }
  }
}
```

See more: https://stackoverflow.com/questions/43281741/how-to-use-paths-in-tsconfig-json

**Comment by nitinpatelm** [comment]
Same problem for me

**Reply by officialkevinbrian** [reply]
@nitinpatelm did you find solution?

**Comment by MMTE** [comment]
check if paths option is available both in tsconfig.json and jsconfig.json if you are using typescript or javascript.

**Comment by alimukhtar53** [comment]
I had to rename my dir name inside components from **stepsInfo** to **steps-info**

My project setup was:

project-root/
├── app/
│   ├── components/
│   │   └── stepsInfo/
│   │       └── Step1.tsx
├── tsconfig.json

**[Reference](https://vercel.com/guides/how-do-i-resolve-a-module-not-found-error#:~:text=The%20'module%20not%20found'%20error%20is%20a%20syntax%20error%20that,and%20local%20machine%2C%20or%20both.)**

**Comment by AleksaRadovanovic** [comment]
You probably renamed folder in components, in my case I renamed components/UI to components/ui. 
Git might not recognize this by default. To fix it, run:

`git config core.ignorecase false`

Then commit and push your changes.

**Reply by chbardamu** [reply]
My case exactly

**Reply by grantimbo** [reply]
Thanks 👏

**Reply by Brundeco** [reply]
Hero!

**Reply by Danieldo1** [reply]
**THIS !!!!**

**Reply by ExoticCoder16** [reply]
This completely solved my problem ! Can't thank you enough !

**Reply by ESSAMMOHAMED1** [reply]
thanks man

**Reply by mickeymaruf** [reply]
**Let's Gooo....👏👏**

**Reply by wleklinskimateusz** [reply]
Wow, that was actually the case, I've forgotten I've even changed the name ;D

**Reply by shobhitexe** [reply]
This one solved my problem Thanks, man 🙌🚀

**Reply by Shubham-Mi** [reply]
This really helped. Thank you! :rocket:

**Reply by skafte-develop** [reply]
Thank you!!

**Reply by euhansarkar** [reply]
thank you sooooo much!

**Reply by MrKou47** [reply]
You saved me today. Thanks!

**Reply by tahsinature** [reply]
Thanks a lot. Saved my day.
Wasn't expecting this. Didn't know git has case insensitivity too.

**Reply by artemkamyshenkov** [reply]
Thank you a lot, bro

**Reply by IsmaelOmar** [reply]
I love you

**Reply by lilsouza** [reply]
Thank you so much!!!

**Reply by benderlidze** [reply]
And you may need to do this after pull
```

//Clear Next.js build cache
rm -rf .next
//Reinstall node_modules
rm -rf node_modules
npm install
// Rebuild the project
npm run build
```

**Reply by sodiqsanusi** [reply]
Thank you so much, works now!

**Reply by imSeptember** [reply]
Yeah man you are good

**Reply by jovimoura** [reply]
i love u bro

**Reply by meraeugene** [reply]
i <3 u

**Reply by arseniypom** [reply]
Just amazing, thank you so much!

**Reply by TheShagwell** [reply]
![Screenshot (107)](https://github.com/user-attachments/assets/84de80d2-a24b-4a13-9d74-3af3d7723d0c)

And yeah you saved the day 😎🤩

**Reply by RodrigoColin1602** [reply]
This really worked for me, i think i had problems due to folders with the same name . Hero

**Reply by parsifal486** [reply]
very very love you!!!

**Reply by yusifaliyevpro** [reply]
Thank you very much

**Comment by fyargat** [comment]
```js
// next.config.js
/** @type {import('next').NextConfig} */

const path = require('path');

const nextConfig = {
  webpack: (config) => {
    config.resolve.alias = {
      ...config.resolve.alias,
      '@': path.resolve(__dirname, './'),
    };

    return config;
  },
};

module.exports = nextConfig;
```

See more: https://stackoverflow.com/questions/76676456/next-js-v13-in-docker-does-not-respect-path-alias-but-works-locally

**Reply by Muskulee** [reply]
This resolves the issue: You can however use the below in ES6:
`import path from 'path';`

`const __dirname = new URL('.', import.meta.url).pathname;`

`const nextConfig = {`
 ` webpack: (config) => {`
    `config.resolve.alias = {`
     ` ...config.resolve.alias,`
      `'@': path.resolve(__dirname, './'),`
   ` }`;
    `return config;`
  `},`
`};`
`export default nextConfig;`

I hope this helps too!

**Reply by mmk0612** [reply]
This worked for me update the next.Config.mjs file this way

**Reply by YoSoyPhil** [reply]
Thank you so much! <3

**Comment by rajdevkumar2003** [comment]
"paths": {
      "@/*": ["./*"]
    }

**Comment by rajdevkumar2003** [comment]
👍🏼

**Comment by ciaranmccaughey** [comment]
I encountered the same issue during next build in my pipelines. Here's a breakdown of the problem and solution:

Problem:
1. I was setting the node environment to prod, `NODE_ENV=production` which prevented dev dependencies from being installed.
2. Without the dev dependencies such as TypeScript, the tsconfig file was not recognised.
3. Consequently, alias paths (e.g., @components/) defined in tsconfig were not found.

Solution:
1. Set `NODE_ENV=development` before the package installs `NODE_ENV=development npm install`
2. Alias paths defined in tsconfig (like @components/) are now correctly resolved.

**Reply by ricoardiya** [reply]
@ciaranmccaughey you just saved my day!

**Comment by ayakutt** [comment]
If you are not using typescript, try deleting tsconfig.json

**Reply by Symbolistic** [reply]
Bro I love you oh my god!
This worked

**Reply by ycenta** [reply]
Real answer that fix the problem 👍

**Reply by shubh95rai** [reply]
it worked! thanks!

**Reply by prosanjeev** [reply]
great bro

**Comment by omkute7** [comment]
**Issue Breakdown : Module not found despite of right path.**

**Solution** :  I was using `jsx `and accidently created `tsx `file for a next page, A new `config,tsx `file got created, this path issue is due to that` config.tsx` file, you can delete it and your issue will be fixed.

**Conslole error :-**
 Creating an optimized production build ...
Failed to compile.

./src/app/layout.js
Module not found: Can't resolve '@/components/ui/NavBar'

.... more of this paths down

**Comment by edtimer** [comment]
For me it was related to a file name case with absolute import. The following removed the cache and deployment worked fine.

`git rm -r --cached .`
`git add --all .`
`git commit -a -m "Versioning untracked files"`
`git push origin master` //main or master based on your repo

**Comment by akakiplimo** [comment]
I encountered a similar issue while working on a Next.js project using Javascript. I accidentally created a file with `.tsx` instead of `.jsx`, this automatically triggered a `next.config.ts` and `tsconfig.json` file to be created which made the issue persist.

**Solution**: Delete the Typescript related files then restart the server using npm run dev your issue should be resolved
